### PR TITLE
fix(vscode): disable extension if required config file is missing

### PR DESCRIPTION
**Pull request by nhedger**
## Summary

This PR fixes one of the two issues discussed in #115.

It prevents the extension from proceeding with the initialization process if the `biome.json` configuration is required by the `biome.requireConfiguration` setting (default) and is missing from the workspace.

If the configuration file is not required, the initialization process will proceed as usual.

This implementation could undoubtedly be a bit more robust, but I will propose additional improvements to the extension in the future. For the time being, this should ensure the extension users aren't being nagged every time they open VS Code.

## Test Plan

- [x] If a configuration file is required and missing, the extension no longer attempts (and fails) to locate the biome server.

### CODE REVIEW

File: editors/vscode/src/main.ts [hunk 1]
```diff
@@ -39,6 +39,26 @@ export async function activate(context: ExtensionContext) {
...
 
+	const requiresConfiguration = workspace
+		.getConfiguration("biome")
+		.get<boolean>("requireConfiguration");
+
+	// If the extension requires a configuration file to be present, we attempt to
+	// locate it. If a config file cannot be found, we do not go any further.
+	if (requiresConfiguration) {
+		outputChannel.appendLine("Configuration file required, looking for one.");
+		const configFiles = await workspace.findFiles("**/biome.json");
```
Comment (ematipico) [inline editors/vscode/src/main.ts:50]: We should also check for `rome.json` file. Even though it's deprecated, it's still supported.

While doing so, could you add a `TODO` comment like `// TODO: remove in biome 2.0`, please?

**Review by ematipico — APPROVED** [review]
It looks good to me; I suggest just a change required for the extension to keep working.

Also, have you tried to test it locally? If you run `npm run pack:dev`, you should be able to test your changes in your VSCode.

File: editors/vscode/src/main.ts (hunk 1 above)
Comment (nhedger) [reply editors/vscode/src/main.ts]: Yep, I forgot about that one. Thanks for pointing it out.

**Comment by nhedger** [comment]
> Also, have you tried to test it locally? If you run npm run pack:dev, you should be able to test your changes in your VSCode.

I did, and it works as expected.

**Comment by nhedger** [comment]
The CI is failing because the formatting is incorrect.

I'm responsible for some of the errors, but there are other formatting issues that I did not introduce. I suspect they were already there.

I guess I'll just run the formatting on the whole vscode directory. Feel free to revert if it does not work for you.

**Pull request closed without merging**